include/zephyr/mgmt/mcumgr/mgmt/mgmt.h

```diff
@@ -178,10 +178,12 @@ typedef int (*mgmt_handler_fn)(struct smp_streamer *ctxt);
 
 /**
  * @brief Read handler and write handler for a single command ID.
+ * Set use_custom_payload to true when using a user defined payload type
  */
 struct mgmt_handler {
 	mgmt_handler_fn mh_read;
 	mgmt_handler_fn mh_write;
+	bool use_custom_payload;
```

Comment:
This increases the default build of a random board I picked of the smp_svr sample by 148 bytes, nothing unacceptable but that is 148 bytes used for basically nothing. Can you therefore add a Kconfig option which is used to enable this (disabled by default), when the Kconfig option is not set then this flag does not exist, when set it exists and is used by the code. Kconfig option can be added to `subsys/mgmt/mcumgr/mgmt/Kconfig`. Out of tree code can select it and have it enabled when it is needed.

Reply:
Good idea, I have added a Kconfig